projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts

```diff
@@ -101,7 +101,18 @@ export class IgxForOfDirective<T> implements OnInit, OnChanges, DoCheck, OnDestr
      * ```
      */
     @Input()
-    public igxForItemSize: any;
+    get igxForItemSize(): any {
+        return this._igxForItemSize;
+    }
+
+    set igxForItemSize(value) {
+        this._igxForItemSize = value;
+        if (this.dc) {
+            this.initSizesCache(this.igxForOf);
+            this._applyChanges();
+            this.onDataChanged.emit();
```

Comment:
Is there a reason onDataChanged needs to be emitted here? It should be emitted only when the actual data changes.

Reply:
Yes, I'll remove it.